generators/chipyard/src/main/scala/ChipTop.scala

```diff
@@ -15,6 +15,9 @@ import barstools.iocell.chisel._
 
 case object BuildSystem extends Field[Parameters => LazyModule]((p: Parameters) => new DigitalTop()(p))
 
+trait HasReferenceClockFreq {
+  var refClockFreqMHz: Option[Double] = None
```

Comment:
I had alot of trouble making this a `def` since the `def` has to be defined within the clock scheme function... Maybe I'm missing something but can you assign a `def` inside a function?

Reply:
You can make the clock scheme function return the freq.
So in ChipTop you could have 
```
val refClockFreqMHz = p(ClockingSchemeKey)(this)
```
`val` can override `def` in Scala.

Reply:
I'm aware of that... I was hitting some snags with the frequency determined in the module body and passing that out of the clock function. I'll try hacking away to find a solution.